test/integration/integration_test.cc

```diff
@@ -159,6 +160,64 @@ TEST_P(IntegrationTest, PerWorkerStatsAndBalancing) {
   check_listener_stats(0, 1);
 }
 
+// Test extend balance.
+TEST_P(IntegrationTest, PerWorkerStatsAndExtendBalancing) {
+  concurrency_ = 2;
+
+  config_helper_.addConfigModifier([&](envoy::config::bootstrap::v3::Bootstrap& bootstrap) -> void {
+    Network::TestConnectionBalanceFactory test_connection_balancer;
+    Registry::InjectFactory<Envoy::Network::ConnectionBalanceFactory> inject_factory(
```

Comment:
I create a balancer for test, it only uses first registered handler.

Seems there is a limitation that I can not inject a new factory during the test, can I add a new extension for the connection balancer(random connection balancer)? Or is there something I miss?

Reply:
Plenty of tests register new extensions so there must be some other issue. I would recommend debugging.

/wait

Reply:
I don't see the REGISTER_FACTORY macro in this PR?

Reply:
https://github.com/envoyproxy/envoy/blob/c0d77b3a4c8cd1225f5b071bf060214ca8309bb5/tools/code_format/check_format.py#L49-L55

It says do not use REGISTER_FACTORY in tests, I also try add this, but not helpful. I will continue to debug.

Reply:
Root cause is wrong type_url, I fake one but invalid.